### DISCUSSION TITLE
read from json or yaml configuration

### DISCUSSION THREAD

**Discussion by vsudhini** [ZiggyCreatures/FusionCache#186] [Ideas]
ability to read cache settings from yaml or json configuration files. set an attribute that can use the configuration entry

**Comment by jodydonetti** [comment]
Hi @vsudhini and thanks for using FusionCache!

I'm not sure I understand what you are asking, but FusionCache can be configured like any other .NET component via the [standard configuration system](https://learn.microsoft.com/en-us/dotnet/core/extensions/configuration), and it also supports the standard [IOptions<T> pattern](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options).

If you want to use a YAML file as the configuration, you can already do this and it's not tied speciically to FusionCache: read [here](https://khalidabuhakmeh.com/top-five-configuration-options-for-dotnet-core) for an example.

Hope this helps, let me know.

**Comment by vsudhini** [comment]
want to do something like this, where I can read named cache settings from config. Wanted to see if something like this can be available out of the box with all options.
{
  "RedisConnectionString": "[YOUR_REDIS_CONNECTION_STRING]",
  "Serializer": "NewtonsoftJson",
  "Caches": {
    "InMemoryOnlyCache": {
      "CacheLocation": "Memory",
      "Backplane": "None", // Use memory-only cache
      "EntryOptions": {
        "Duration": "00:01:00" // 1 minute
      }
    },
    "MyRedisCache": {
      "CacheLocation": "Redis",
      "Backplane": "Redis", // Use Redis as backplane
      "EntryOptions": {
        "Duration": "00:05:00" // 5 minutes
      }
    }
  }
}

**Reply by jodydonetti** [reply]
Some of these concerns are more "configurations" (meaning declarative), others are more "setup" (meaning imperative): to go down this road I should define all the keys for all the "extensions" like types of distributed cache, types of plugins, etc and each dependency can have different ways to configure them, so to define a unified solution for this is highly unlikely.
I don't think something like this should be in the core project: on the other hand I think it can be easily made as a custom piece of code that would load the dependencies and set them up in your own way.
